### PR TITLE
irb: Use Parser to parse args

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
#1860

### CODE REVIEW

File: Library/Homebrew/dev-cmd/irb.rb [hunk 1]
```diff
@@ -21,15 +24,20 @@ module Homebrew
...
+    if args.examples?
       puts "'v8'.f # => instance of the v8 formula"
       puts ":hub.f.installed?"
       puts ":lua.f.methods - 1.methods"
       puts ":mpd.f.recursive_dependencies.reject(&:installed?)"
       return
     end
 
-    if ARGV.pry?
+    if args.pry? || Utils::EnvVars.pry?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/irb.rb]: Is it possible for `args.pry?` to also handle the environment variable somehow? Also, you can probably delete https://github.com/Homebrew/brew/blob/121780b41701d1d4857b65c4a3f0ecd511807f75/Library/Homebrew/extend/ARGV.rb#L271-L273 now (and in similar other PRs).

File: Library/Homebrew/dev-cmd/irb.rb [hunk 2]
```diff
@@ -21,15 +24,20 @@ module Homebrew
   module_function
 
   def irb
-    if ARGV.include? "--examples"
+    args = Homebrew::CLI::Parser.new do
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/irb.rb]: I was just thinking while reading this: what do you think about a refactoring that looks like this:
```ruby
    args = Homebrew::CLI::Parser.parse do
      switch "--examples"
      switch "--pry"
    end
```
or similar as you never seem to make use of the `Parser` object directly.

File: Library/Homebrew/dev-cmd/irb.rb (hunk 1 above)
Comment (GauthamGoli) [reply Library/Homebrew/dev-cmd/irb.rb]: For `args.OPTION?` to handle environment variables, would probably have to assume that env var naming has the naming convention `HOMEBREW_<OPTION>`

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/irb.rb]: Yeh, that would work. Additionally, we could allow them to be specified with the DSL e.g. `env: :pry` or similar to mean `HOMEBREW_PRY` can trigger it.

Comment (GauthamGoli) [reply Library/Homebrew/dev-cmd/irb.rb]: Explicitly specifying in DSL seems better to me.

File: Library/Homebrew/dev-cmd/irb.rb (hunk 2 above)
Comment (GauthamGoli) [reply Library/Homebrew/dev-cmd/irb.rb]: Yeah, seems better & more readable this way. Will open a seperate PR for this.

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/irb.rb]: @GauthamGoli Cool. Feel free to combine both ideas above into same PR (although different commits ideally).

Comment (GauthamGoli) [reply Library/Homebrew/dev-cmd/irb.rb]: @MikeMcQuaid I've added option for specifying environment variables and while adding a test for this, I remembered that I do make use of `Parser` Object directly in tests. It was written in existing way to make testing easier.

**Comment by MikeMcQuaid** [comment]
Thanks again @GauthamGoli, nice work!

File: Library/Homebrew/dev-cmd/irb.rb (hunk 2 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/irb.rb]: @GauthamGoli Should still be able to have a `.parse` class method which calls `new` and then runs `parse` on it so it's a short-hand of the version here without needing to change the existing `new` or `#parse` methods. As a general principle I'd say a public API should never be less readable/harder to use just to make testing easier.

Comment (GauthamGoli) [reply Library/Homebrew/dev-cmd/irb.rb]: Agree. Will do it. 👍

**Pull request merged**